_posts/2017-09-05-planning-for-7.0.md

```diff
@@ -0,0 +1,610 @@
+let generator = _skipFirstGeneratorNext(function* () {
+    const _functionSent = yield;
+    console.log("Sent", _functionSent);
+    console.log("Yield", yield);
+});
+```
+
+### Stage 2: export-ns-from
+
+> `babel-plugin-transform-export-namespace`: a shorthand to import/reexport a namespace. Split out from the old `transform-export-extensions` which combined this proposal with another
+
+
+Input
+
+```js
+export * as ns from "mod";
+```
+
+Output
+
+```js
+import * as ns from "mod";
+export {ns};
+```
+
+### Stage 1: export-default-from
+
+> `babel-plugin-transform-export-default`: a shorthand to import/reexport something. Split out from the old `transform-export-extensions` which combined this proposal with another
+
+
+Input
+
+```js
+export v from "mod";
+```
+
+Output
+
+```js
+import _v from "module";
+export { _v as v };
+```
+
+### Stage 1: Optional Chaining (new)
+
+> `babel-plugin-transform-optional-chaining`: the operator (`?.`) allows you to handle properties of deeply nested objects without worrying about undefined intermediate objects.
+
+Input
+
+```js
+a?.b = 42;
+```
+
+Output
+
+```js
+var _a;
+(_a = a) == null ? void 0 : _a.b = 42;
+```
+
+### ES2015: `new.target`
+
+> `babel-plugin-transform-new-target`: we never got around to implementing `new.target` support earlier, so now there is a plugin for it that will be included in the ES2015/env presets.
+
+Example
+
+```js
+// with a function
+function Foo() {
+  console.log(new.target);
+}
+
+Foo(); // => undefined
+new Foo(); // => Foo
+
+// with classes
+class Foo {
+  constructor() {
+    console.log(new.target);
+  }
+}
+
+class Bar extends Foo {
+}
+
+new Foo(); // => Foo
+new Bar(); // => Bar
+```
+
+Input
+
+```js
+class Foo {
+  constructor() {
+    new.target;
+  }
+
+  test() {
+    new.target;
+  }
+}
+```
+
+Output
+
+```js
+class Foo {
+  constructor() {
+    this.constructor;
+  }
+
+  test() {
+    void 0;
+  }
+}
+```
+
+## 🚀 New Feature
+
+### `.babelrc.js`
+
+> [babel/babel#4630](https://github.com/babel/babel/issues/4630)
+
+`*.js` configuration files are fairly common in the JavaScript ecosystem. ESLint and Webpack both allow for `.eslintrc.js` and `webpack.config.js` configuration files, respectively.
+
+Writing configuration files in JavaScript allows for dynamic configuration, making it possible to write a single configuration file that can adapt to different environments programmatically.
+
+```js
+var env = process.env.BABEL_ENV || process.env.NODE_ENV;
+var plugins = [];
+if (env === 'production') {
+  plugins.push.apply(plugins, ["a-super-cool-babel-plugin"]);
+}
+module.exports = { plugins };
+```
+
+```js
+var env = process.env.BABEL_ENV || process.env.NODE_ENV;
+module.exports = {
+  plugins: [
+    env === 'production' && "another-super-cool-babel-plugin"
+  ].filter(Boolean)
+};
+```
+
+This was previously done through the `env` configuration option, which is now deprecated. See [below](#deprecate-the-env-option-in-babelrc) for more details.
```

Comment:
Probably good to mention the babel 6 forwards-compatible option of having a `{ "presets": ["./.babelrc.js"] }` .babelrc